### ISSUE TITLE
Image buffer displays outside of embed

### ISSUE THREAD

**Issue by jamboj** [discordjs/guide#242]
It looks like there is no way to have image buffer display in embed properly - it displays outside and not inside. What's worse it messes up notifications that just become "uploaded file.jpg"  instead of the title/description. 

To reproduce:
```
var img = await Jimp.read('https://i.imgur.com/wSTFkRM.png');
var imgBuf = await img.getBufferAsync(Jimp.AUTO);

const exampleEmbed = new Discord.RichEmbed()
.setTitle('important title')
.setDescription('meaningful description')
.attachFiles([imgBuf])
.setImage('attachment://imgBuf');

discordChannel.send(exampleEmbed);
```
results in: 

![image](https://user-images.githubusercontent.com/36251521/60683604-9da1f300-9e66-11e9-86cb-35362f1e9741.png)

![image](https://user-images.githubusercontent.com/36251521/60683620-bc07ee80-9e66-11e9-9917-b9da0e3bc402.png)

**Comment by almostSouji** [comment]
As you can see [here](https://discord.js.org/#/docs/main/stable/class/RichEmbed?scrollTo=attachFiles) attachFiles takes an array of [FileOptions](https://discord.js.org/#/docs/main/stable/class/RichEmbed?scrollTo=attachFiles) which may include a name and the attachment itself.
Naming the file then allows you to properly reference it from the `setImage` method, which I hoped the yellow warnings all over the guide would suggest.

There's also quite some redundancy in your snippet as you can directly provide an URL as buffer to above mentioned attachment property.
If you have the image on imgur anyways reattaching it is however not even needed as you can simply set embed images to an URL directly.

**Comment by jamboj** [comment]
Thank you @almostSouji 
This works:

```
var img = await Jimp.read('https://i.imgur.com/wSTFkRM.png');
var imgBuf = await img.getBufferAsync(Jimp.AUTO);

const exampleEmbed = new Discord.RichEmbed()
.setTitle('important title')
.setDescription('meaningful description')
.attachFiles([{name: "image.png", attachment:imgBuf}])
.setImage('attachment://image.png');

discordChannel.send(exampleEmbed);
```
However this doesn't work if you don't use '.png' at the end of 'file' name and it also should not just give a crooked output if used incorrectly.

Re redundancy - I think you missed my point, my code snippet is just using a file but the point is to illustrate work with a local buffer. In real life application one might not have a file.

I did see the warnings and I googled far and deep and did try every possible combination I could think of before opening this issue, so I think you might need to update the code or at least add an example of working with buffer here: https://discordjs.guide/popular-topics/embeds.html#using-the-richembed-constructor

otherwise it seems like really hard to discover. Thanks again for your prompt reply and giving a hint that finally solved my issue
